Google.PowerShell/PubSub/GcpsMessage.cs

```diff
@@ -0,0 +1,222 @@
+            }
+            PubsubMessage psMessage = new PubsubMessage() { Data = data, Attributes = attributesDict };
+            return psMessage;
+        }
+    }
+
+    /// <summary>
+    /// <para type="synopsis">
+    /// Publishes one or more PubSub messages to a topic.
+    /// </para>
+    /// <para type="description">
+    /// Publishes one or more PubSub messages to a topic. Will raise errors if the topic does not exist.
+    /// The cmdlet will search for the topic in the default project if -Project is not used.
+    /// To publish more than 1 message, use -Message parameter with an array of messages constructed from New-GcpsMessage.
+    /// Otherwise, use -Data and -Attribute parameters to publish a single message.
+    /// </para>
+    /// <example>
+    ///   <code>PS C:\> Publish-GcpsTopic -Topic "my-topic" -Data "This is a test." -Attributes @{"key"="value"}</code>
+    ///   <para>
+    ///   This command publishes a message with the specified data and attribute to topic "my-topic" in the default project.
+    ///   </para>
+    /// </example>
+    /// <example>
+    ///   <code>
+    ///   PS C:\> $message1 = New-GcpsMessage -Data "my-data"
+    ///   PS C:\> $message2 = New-GcpsMessage -Data "my-data2" -Attributes @{"key"="test"}
+    ///   PS C:\> Publish-GcpsTopic -Topic "my-topic" -Message $message1, $message2 -Project "my-project"
+    ///   </code>
+    ///   <para>
+    ///   This command publishes 2 messages to topic "my-topic" in the project "my-project".
+    ///   </para>
+    /// </example>
+    /// <para type="link" uri="(https://cloud.google.com/pubsub/docs/publisher#publish-messages-to-a-topic)">
+    /// [Publishing Messages to a Topic]
+    /// </para>
+    /// </summary>
+    [Cmdlet(VerbsData.Publish, "GcpsMessage", DefaultParameterSetName = ParameterSetNames.DataAndAttributes)]
+    public class PublishGcpsMessage : GcpsCmdlet
+    {
+        private class ParameterSetNames
+        {
+            public const string DataAndAttributes = "DataAndAttributes";
+            public const string Message = "Message";
+        }
+
+        /// <summary>
+        /// <para type="description">
+        /// The project to check for topic. If not set via PowerShell parameter processing, will
+        /// default to the Cloud SDK's DefaultProject property.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = false)]
+        [ConfigPropertyName(CloudSdkSettings.CommonProperties.Project)]
+        public string Project { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// The topic to which the messages will be published.
+        /// </para>
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0)]
+        [ValidateNotNullOrEmpty]
+        public string Topic { get; set; }
+
+        /// <summary>
+        /// <para type="description">
+        /// Data of the message that will be published.
```

Comment:
"Data of the message.." this seems a little hard to parse.  Maybe "The data message.."?